include/NovelRT/Animation/SpriteAnimatorState.h

```diff
@@ -0,0 +1,70 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT License (MIT). See LICENCE.md in the repository root for more information.
+
+#ifndef NOVELRT_ANIMATION_SPRITEANIMATORSTATE_H
+#define NOVELRT_ANIMATION_SPRITEANIMATORSTATE_H
+
+#ifndef NOVELRT_H
+#error Please do not include this directly. Use the centralised header (NovelRT.h) instead!
+#endif
+
+namespace NovelRT::Animation {
+  class SpriteAnimatorState {
+  private:
+    std::vector<SpriteAnimatorFrame> _frames;
+    bool _shouldLoop;
+    std::vector<std::tuple<std::shared_ptr<SpriteAnimatorState>, std::vector<std::function<bool()>>>> _transitions;
```

Comment:
if we renamed `Event` to `Delegate`, we could use it in more places (like the `std::vector<std::function<bool()>>` here).

Reply:
(just a note for a potential future cleanup)